tests/unit/utils/test_ratelimiter.py

```diff
@@ -0,0 +1,110 @@
+import time
+from datetime import datetime
+from unittest.mock import AsyncMock, patch
+
+import pytest
+from freezegun import freeze_time
+
+from deepset_cloud_sdk._utils.ratelimiter import RateLimiter
+
+pytest_plugins = ("pytest_asyncio",)
+
+
+# These tests need timeouts, if there is a bug in RateLimiter, it could result in the tests running forever
+@pytest.mark.timeout(0.1)
+class TestRateLimiter:
+    @pytest.fixture
+    def client_mock(self) -> AsyncMock:
+        return AsyncMock()
+
+    @pytest.mark.asyncio
+    async def test_ratelimiter_consumes_tokens(self, client_mock: AsyncMock) -> None:
+        with freeze_time(datetime.now()) as _:
+            ratelimiter = RateLimiter(client=client_mock)
+            ratelimiter._updated_at = time.monotonic()
```

Comment:
why do we need to set this? It's already set by the constructor, isn't it?

Reply:
ah, true!